Self-Hosted Gitlab with API-Token: 401 Unauthorized

When I try to run `todocheck` on our private gitlab instance (`13.2.3`), I get the following failed result:
```bash
$ export TODOCHECK_AUTH_TOKEN=[MY-PERSONAL-ACCESS-TOKEN]
$ ~/Schreibtisch/todocheck-0.2.0-linux-x86_64 --config .todocheck.yaml --basepath src/
2020/08/14 17:08:05 couldn't traverse basepath: failed traversing file src/Some/Example/File.php: couldn't check todo line: couldn't fetch task status: bad status code upon fetching task: 401 - {"message":"401 Unauthorized"}
$ 
```

When I try to manually fetch the issue-details from the gitlab-API using `curl`, I get this successful result:
```bash
$ curl --header "PRIVATE-TOKEN: [MY-PERSONAL-ACCESS-TOKEN]" "https://our-gitlab-instance.com/api/v4/projects/3/issues/1"
{"id":16,"iid":1,"project_id":3,"title":"Some Title","description":"Lorem ipsum dolor sit amet.","state":"opened","created_at":"2020-08-14T14:47:36.336Z","updated_at":"2020-08-14T14:47:36.336Z","closed_at":null,"closed_by":null,"labels":[],"milestone":null,"assignees":[],"author":{"id":6,"name":"Gerrit Addiks","username":"gerrit","state":"active","avatar_url":"https://our-gitlab-instance.com/uploads/-/system/user/avatar/6/avatar.png","web_url":"https://our-gitlab-instance.com/gerrit"},"assignee":null,"user_notes_count":0,"merge_requests_count":0,"upvotes":0,"downvotes":0,"due_date":null,"confidential":false,"discussion_locked":null,"web_url":"https://our-gitlab-instance.com/foo/bar/-/issues/1","time_stats":{"time_estimate":0,"total_time_spent":0,"human_time_estimate":null,"human_total_time_spent":null},"task_completion_status":{"count":0,"completed_count":0},"has_tasks":false,"_links":{"self":"https://our-gitlab-instance.com/api/v4/projects/3/issues/1","notes":"https://our-gitlab-instance.com/api/v4/projects/3/issues/1/notes","award_emoji":"https://our-gitlab-instance.com/api/v4/projects/3/issues/1/award_emoji","project":"https://our-gitlab-instance.com/api/v4/projects/3"},"references":{"short":"#1","relative":"#1","full":"foo/bar#1"},"subscribed":true,"moved_to_id":null}
$
```

My `.todocheck.yaml` file:
```yaml
origin: https://our-gitlab-instance.com/foo/bar
issue_tracker: GITLAB
auth:
  type: apitoken
```

Why is `todocheck` unable to fetch the issue while `curl` can, using the same access-token?
Is there maybe some way to enable debugging and/or to see what exact HTTP(S) requests are being sent?

When I try to download the binary I am asked for a password for that file: "You need access, Ask for access, or switch to an account with access".

I also tried to compile it from source, but got stuck on these error-messages:
```bash
$ ./release.sh bin/ 0.2.0
main.go:10:2: cannot find package "github.com/preslavmihaylov/todocheck/authmanager" in any of:
	/usr/lib/go-1.6/src/github.com/preslavmihaylov/todocheck/authmanager (from $GOROOT)
	($GOPATH not set)
main.go:11:2: cannot find package "github.com/preslavmihaylov/todocheck/authmanager/authmiddleware" in any of:
	/usr/lib/go-1.6/src/github.com/preslavmihaylov/todocheck/authmanager/authmiddleware (from $GOROOT)
	($GOPATH not set)
main.go:12:2: cannot find package "github.com/preslavmihaylov/todocheck/checker/errors" in any of:
	/usr/lib/go-1.6/src/github.com/preslavmihaylov/todocheck/checker/errors (from $GOROOT)
	($GOPATH not set)
main.go:13:2: cannot find package "github.com/preslavmihaylov/todocheck/config" in any of:
	/usr/lib/go-1.6/src/github.com/preslavmihaylov/todocheck/config (from $GOROOT)
	($GOPATH not set)
main.go:14:2: cannot find package "github.com/preslavmihaylov/todocheck/fetcher" in any of:
	/usr/lib/go-1.6/src/github.com/preslavmihaylov/todocheck/fetcher (from $GOROOT)
	($GOPATH not set)
main.go:15:2: cannot find package "github.com/preslavmihaylov/todocheck/issuetracker" in any of:
	/usr/lib/go-1.6/src/github.com/preslavmihaylov/todocheck/issuetracker (from $GOROOT)
	($GOPATH not set)
main.go:16:2: cannot find package "github.com/preslavmihaylov/todocheck/traverser/todoerrs" in any of:
	/usr/lib/go-1.6/src/github.com/preslavmihaylov/todocheck/traverser/todoerrs (from $GOROOT)
	($GOPATH not set)
shasum: todocheck-0.2.0-windows-x86_64.exe: 
can't load package: package .: no buildable Go source files in /home/gerrit/git/todocheck/bin
./release.sh: Zeile 21: cd: bin/: Datei oder Verzeichnis nicht gefunden
can't load package: package .: no buildable Go source files in /home/gerrit/git/todocheck/bin
./release.sh: Zeile 21: cd: bin/: Datei oder Verzeichnis nicht gefunden
can't load package: package .: no buildable Go source files in /home/gerrit/git/todocheck/bin
./release.sh: Zeile 21: cd: bin/: Datei oder Verzeichnis nicht gefunden
$
```
(`Datei oder Verzeichnis nicht gefunden` = `File not found`)

Since i'm really not proficient with go and a bit of research suggests that issues with GOPATH might be version-related (I'm on ubuntu 16.04, installing newer software can sometimes be non-trivial), I don't think that dumping more time into this will yield any results.

--- a/authmanager/authmanager.go
+++ b/authmanager/authmanager.go
@@ -8,13 +8,14 @@ import (
 	"syscall"
 
 	"github.com/preslavmihaylov/todocheck/authmanager/authstore"
+	"github.com/preslavmihaylov/todocheck/common"
 	"github.com/preslavmihaylov/todocheck/config"
 	"golang.org/x/crypto/ssh/terminal"
 )
 
 const (
 	githubAPITokenMsg  = "Please go to https://github.com/settings/tokens, create a read-only access token & paste it here:\nToken: "
-	gitlabAPITokenMsg  = "Please go to https://gitlab.com/profile/personal_access_tokens, create a read-only access token & paste it here:\nToken: "
+	gitlabAPITokenMsg  = "Please go to %s/profile/personal_access_tokens, create a read-only access token & paste it here:\nToken: "
 	pivotalAPITokenMsg = "Please go to https://www.pivotaltracker.com/profile, create a new API token & paste it here:\nToken: "
 	redmineAPITokenMsg = "Please go to %s/my/account, create a new API token & paste it here:\nToken: "
 
@@ -41,7 +42,7 @@ func acquireAPIToken(cfg *config.Local) error {
 		if cfg.IssueTracker == config.IssueTrackerGithub {
 			msg = githubAPITokenMsg
 		} else if cfg.IssueTracker == config.IssueTrackerGitlab {
-			msg = gitlabAPITokenMsg
+			msg = fmt.Sprintf(gitlabAPITokenMsg, extractBaseURL(cfg.Origin))
 		} else if cfg.IssueTracker == config.IssueTrackerPivotal {
 			msg = pivotalAPITokenMsg
 		} else if cfg.IssueTracker == config.IssueTrackerRedmine {
@@ -99,4 +100,13 @@ func setAndPersistToken(authCfg *config.Auth, store *authstore.Config, key, toke
 	authCfg.Token = token
 	store.Tokens[key] = authCfg.Token
 	return store.Save(authCfg.TokensCache)
+}
+
+func extractBaseURL(origin string) string {
+	tokens := common.RemoveEmptyTokens(strings.Split(origin, "/"))
+	if tokens[0] != "http:" && tokens[0] != "https:" {
+		return fmt.Sprintf("https://%s", tokens[0])
+	}
+
+	return fmt.Sprintf("%s//%s", tokens[0], tokens[1])
 }

--- a/issuetracker/issuetracker.go
+++ b/issuetracker/issuetracker.go
@@ -56,13 +56,13 @@ func BaseURLFor(issueTracker config.IssueTracker, origin string) (string, error)
 		return fmt.Sprintf("%s//api.github.com/repos/%s/%s/issues/", scheme, owner, repo), nil
 	case config.IssueTrackerGitlab:
 		tokens := common.RemoveEmptyTokens(strings.Split(origin, "/"))
-		if tokens[0] == "gitlab.com" {
+		if tokens[0] != "http:" && tokens[0] != "https:" {
 			tokens = append([]string{"https:"}, tokens...)
 		}
 
-		scheme, owner, repo := tokens[0], tokens[2], tokens[3]
+		scheme, host, owner, repo := tokens[0], tokens[1], tokens[2], tokens[3]
 		urlEncodedProject := url.QueryEscape(fmt.Sprintf("%s/%s", owner, repo))
-		return fmt.Sprintf("%s//gitlab.com/api/v4/projects/%s/issues/", scheme, urlEncodedProject), nil
+		return fmt.Sprintf("%s//%s/api/v4/projects/%s/issues/", scheme, host, urlEncodedProject), nil
 	case config.IssueTrackerPivotal:
 		tokens := common.RemoveEmptyTokens(strings.Split(origin, "/"))
 		if tokens[0] == "pivotaltracker.com" {

